From 65bac42e2319bb8d213d188a8ab890f11f576bc8 Mon Sep 17 00:00:00 2001
From: Kostadin Shishmanov <kostadinshishmanov@protonmail.com>
Date: Thu, 13 Mar 2025 13:08:41 +0200
Subject: [PATCH] Add #include <stdint.h> to fix building with gcc 15

With gcc 15, the C++ Standard Library no longer includes other headers
that were internally used by the library. In herbstluftwm's case the
missing header is <stdint.h>

Downstream Gentoo bug: https://bugs.gentoo.org/937529

Closes: #1612
Signed-off-by: Kostadin Shishmanov <kostadinshishmanov@protonmail.com>
---
 src/xconnection.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xconnection.cpp b/src/xconnection.cpp
index e2129db35..788aedfbb 100644
--- a/src/xconnection.cpp
+++ b/src/xconnection.cpp
@@ -6,6 +6,7 @@
 #include <X11/Xutil.h>
 #include <X11/extensions/Xrender.h>
 #include <fcntl.h>
+#include <stdint.h>
 #include <unistd.h>
 #include <climits>
 #include <cstring>
